src/line_ending.rs

```diff
@@ -0,0 +1,117 @@
+//! Line ending detection and conversion.
+
+use std::fmt::{Debug, Formatter};
+use std::str::FromStr;
+
+/// Supported line endings. Like in the Rust's standard library, two
+/// line endings are supported: `\r\n` and `\n`
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum LineEnding {
+    /// _Carriage return and line feed_ – a line ending sequence
+    /// historically used in _Windows_. Corresponds to the sequence
+    /// of ASCII control characters `0x0D 0x0A` or `\r\n`
+    CRLF,
+    /// _Line feed_ – a line ending historically used in _Unix_.
+    ///  Corresponds to the ASCII control character `0x0A` or `\n`
+    LF,
+}
+
+/// Returned when attempted creating [`LineEnding`] value from an
+/// unsupported `&str` value
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub struct UnsupportedLineEnding;
+
+impl std::fmt::Display for UnsupportedLineEnding {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(f, "unsupported line ending sequence")
+    }
+}
+
+impl std::error::Error for UnsupportedLineEnding {}
+
+impl LineEnding {
+    /// Turns this [`LineEnding`] value into its ASCII representation.
+    #[inline]
+    pub const fn as_str(&self) -> &'static str {
+        match self {
+            Self::CRLF => "\r\n",
+            Self::LF => "\n",
+        }
+    }
+}
+
+impl FromStr for LineEnding {
```

Comment:
This is not used and I don't think we need it. Please drop it until someone has a use case for it :-)